digdag-standards/src/main/java/io/digdag/standards/operator/pg/PgConnection.java

```diff
@@ -0,0 +1,193 @@
+    protected static PgConnection open(PgConnectionConfig config)
+    {
+        return new PgConnection(config.openConnection());
+    }
+
+    protected PgConnection(Connection connection)
+    {
+        super(connection);
+    }
+
+    @Override
+    public void executeReadOnlyQuery(String sql, Consumer<JdbcResultSet> resultHandler)
+        throws NotReadOnlyException
+    {
+        try {
+            execute("SET TRANSACTION READ ONLY");
+            try (Statement stmt = connection.createStatement()) {
+                ResultSet rs = stmt.executeQuery(sql);  // executeQuery throws exception if given query includes multiple statements
+                resultHandler.accept(new PgResultSet(rs));
+            }
+            execute("SET TRANSACTION READ WRITE");
+        }
+        catch (SQLException ex) {
+            if (ex.getSQLState().equals("25006")) {  // 25006 = read_only_sql_transaction error
+                throw new NotReadOnlyException(ex);
+            }
+            else {
+                throw new DatabaseException("Failed to execute given SELECT statement", ex);
+            }
+        }
+    }
+
+    @Override
+    public String escapeIdent(String ident)
+    {
+        try {
+            StringBuilder buf = new StringBuilder();
+            escapeIdentifier(buf, ident);
+            return buf.toString();
+        }
+        catch (SQLException ex) {
+            throw new IllegalArgumentException(
+                    String.format(ENGLISH,
+                        "Invalid identifier name (%s): %s",
+                        ex.getMessage(),
+                        ident));
+        }
+    }
+
+    @Override
+    public TransactionHelper getStrictTransactionHelper(String statusTableName, Duration cleanupDuration)
+    {
+        return new PgPersistentTransactionHelper(statusTableName, cleanupDuration);
+    }
+
+    private class PgPersistentTransactionHelper
+            extends AbstractPersistentTransactionHelper
+    {
+        PgPersistentTransactionHelper(String statusTableName, Duration cleanupDuration)
+        {
+            super(statusTableName, cleanupDuration);
+        }
+
+        @Override
+        public void prepare()
+        {
+            executeStatement("create a status table " + escapeIdent(statusTableName),
+                    String.format(ENGLISH,
+                        "CREATE TABLE IF NOT EXISTS %s" +
+                        " (query_id text NOT NULL UNIQUE, created_at timestamptz NOT NULL, completed_at timestamptz)",
+                        escapeIdent(statusTableName))
+                   );
+        }
+
+        @Override
+        public void cleanup()
+        {
+            executeStatement("delete old query status rows from " + escapeIdent(statusTableName) + " table",
+                    String.format(ENGLISH,
+                        "DELETE FROM %s WHERE query_id = ANY(" +
+                        "SELECT query_id FROM %s WHERE completed_at < now() - interval '%d seconds'" +
```

Comment:
The original code https://github.com/treasure-data/digdag/pull/175/commits/95f2bd0dfd9e8d5d25043152a9566fe6136528e0#diff-6796cda912b3d95467f3864be5e93189L106 was `interval '%d' seconds` not `interval '%d' second` and PostgreSQL didn't accept the syntax. So I made the change. But I should've changed that to `interval '%d' second` thinking of compatibility with SQL standard... I'll change it again.